pkgs/desktops/cinnamon/cjs/default.nix

```diff
@@ -33,10 +33,10 @@ stdenv.mkDerivation rec {
     pkg-config
     which # for locale detection
     libxml2 # for xml-stripblanks
+    gobject-introspection
   ];
 
   buildInputs = [
-    gobject-introspection
```

Comment:
a typelib is installed

Reply:
Only a private one that will not be picked up by the setup hook. And the hook is not propagated either.

Reply:
Yeah, that is correct. gi probably needs to be in both here.

Reply:
```diff
diff --git a/pkgs/desktops/cinnamon/cjs/default.nix b/pkgs/desktops/cinnamon/cjs/default.nix
index 681d6afd4ead..1b3aa41221b1 100644
--- a/pkgs/desktops/cinnamon/cjs/default.nix
+++ b/pkgs/desktops/cinnamon/cjs/default.nix
@@ -33,10 +33,13 @@ stdenv.mkDerivation rec {
     pkg-config
     which # for locale detection
     libxml2 # for xml-stripblanks
-    gobject-introspection
+    dbus
+    glib
   ];
 
+  strictDeps = true;
   buildInputs = [
+    gobject-introspection # for library
     cairo
     readline
     spidermonkey_102
```

doesn't fail, can't test cross because it fails `meson.build:261:22: ERROR: Can not run test applications in this cross environment.`

Reply:
added mesonEmulatorHook and not it fails with g-ir-scanner

Reply:
gobject-introspection in nativeBuildInputs already propagates itself to buildInputs, so no need to have it in both, which is why i would prefer having it in nativeBuildInputs always because then cross would always just work

Reply:
I generally prefer explicit direct dependencies over propagation since it paints a clearer picture. But cjs is not my package so I will leave it up to its maintainers.

Reply:
> because then cross would always just work

if having it twice breaks a particular usecase then let's just have it in nativeBuildInputs

otherwise whatever @bobby285271 prefers